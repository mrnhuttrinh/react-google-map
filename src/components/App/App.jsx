import React from 'react';
import { block } from 'bem-cn';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { Header, MainBody } from '../';;

import './App.scss';

const b = block('app-container');

class App extends React.PureComponent {

  render() {
    return (
      <Router>
        <div className={b()}>
          <Header />
          <Switch>
            <Route exact path="/" component={MainBody} />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;