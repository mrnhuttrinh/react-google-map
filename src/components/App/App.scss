.app-container {
  font-family: segoe ui;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  background-color: #f9f9f9;
}