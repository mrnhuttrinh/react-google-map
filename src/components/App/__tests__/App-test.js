import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'

import App from '../App';

const mockStore = configureStore();
const initialState = {
  categories: [],
  stores: [],
  ui: {
    showAll: false,
  },
};

test('should render App component', () => {
  const store = mockStore(initialState)
  const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  ).toJSON();
  expect(component).toMatchSnapshot();
});