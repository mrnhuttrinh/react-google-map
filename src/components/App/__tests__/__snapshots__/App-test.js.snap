// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
<div
  className="app-container"
>
  <header
    className="header"
  />
  <div
    className="store"
  />
</div>
`;
